cli/cmd/deploy/deploy.go

```diff
@@ -99,6 +112,26 @@ func DeployCmd(cfg *config.Config) *cobra.Command {
 				return err
 			}
 
+			// Run flow for access to the Github remote (if necessary)
+			ghRes, err := githubFlow(ctx, client, githubURL, silentGitFlow)
+			if err != nil {
+				return fmt.Errorf("failed Github flow: %w", err)
+			}
+
+			if prodBranch == "" {
+				prodBranch = ghRes.DefaultBranch
+			}
+
+			if !repoInSyncFlow(projectPath, prodBranch) {
+				warn.Println("You can run `rill deploy` again once local changes are added to remote repo.")
+				return nil
+			}
```

Comment:
If user hasn't provided any branch we rely on github repo's default branch which will only be available once user gives access to the repository.

Reply:
The default branch can be extracted directly from the local Git folder. It might not always be the same, but almost always.

Just feels like this is an error people will see quite often, so it's weird to throw it in the middle of the flow.